src/dtls13.c

```diff
@@ -0,0 +1,1000 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#include <wolfssl/wolfcrypt/settings.h>
+
+#ifdef WOLFSSL_DTLS13
+
+#include <wolfssl/ssl.h>
+#include <wolfssl/internal.h>
+#include <wolfssl/error-ssl.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/wolfcrypt/types.h>
+#include <wolfssl/wolfcrypt/aes.h>
+#include <wolfssl/wolfcrypt/kdf.h>
+
+#ifdef NO_INLINE
+    #include <wolfssl/wolfcrypt/misc.h>
+#else
+    #define WOLFSSL_MISC_INCLUDED
+    #include <wolfcrypt/src/misc.c>
+#endif
+
+enum rn_direction {
+    PROTECT = 0,
+    DEPROTECT,
+};
+
+struct dtls13_handshake_header {
+    uint8_t msg_type;
+    byte length[3];
+    byte message_seq[2];
+    byte fragment_offset[3];
+    byte fragment_length[3];
+};
+
+struct dtls13_record_plaintext_header {
+    byte content_type;
+    ProtocolVersion legacy_version_record;
+    byte epoch[2];
+    byte sequence_number[6];
+    byte length[2];
+};
+
+/* No CID, 16bit Seq number, Length present */
+struct dtls13_record_ciphertext_header {
+    /* 0 0 1 C S L E E */
+    byte unified_hdr_flags;
+    byte sequence_number[2];
+    byte length[2];
+};
+
+#define MAX_SEQ_SIZE 2
+#define LEN_SIZE 2
+#define RN_MASK_SIZE 16
+#define HEADER_FLAGS_SIZE 1
+#define SEQ_16_LEN 2
+#define SEQ_8_LEN 1
+#define SEQ_FULL_SIZE 4
+
+#define FIXED_BITS_MASK (0x111 << 5)
+#define FIXED_BITS (0x1 << 5)
+#define C_BIT (0x1 << 4)
+#define S_BIT (0x1 << 3)
+#define L_BIT (0x1 << 2)
+
+WOLFSSL_METHOD* wolfDTLSv1_3_client_method_ex(void* heap)
+{
+    WOLFSSL_METHOD* method;
+
+    WOLFSSL_ENTER("DTLSv1_3_client_method_ex");
+
+    method =  (WOLFSSL_METHOD*) XMALLOC(sizeof(WOLFSSL_METHOD),
+                                  heap, DYNAMIC_TYPE_METHOD);
+    if (method)
+        InitSSL_Method(method, MakeDTLSv1_3());
+
+    return method;
+}
+
+WOLFSSL_METHOD* wolfDTLSv1_3_server_method_ex(void* heap)
+{
+    WOLFSSL_METHOD* method;
+
+    WOLFSSL_ENTER("DTLSv1_3_server_method_ex");
+
+    method = (WOLFSSL_METHOD*) XMALLOC(sizeof(WOLFSSL_METHOD),
+                                  heap, DYNAMIC_TYPE_METHOD);
+    if (method) {
+            InitSSL_Method(method, MakeDTLSv1_3());
+            method->side = WOLFSSL_SERVER_END;
+    }
+
+    return method;
+}
+static int dtls13_rl_add_plaintext_header(WOLFSSL *ssl,
```

Comment:
Formatting
If it's going to squeeze parameters over to the right like that, instead start the next lines with 4 spaces and then the parameters.